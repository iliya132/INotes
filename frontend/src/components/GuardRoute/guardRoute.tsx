import React from 'react';
import { Navigate } from 'react-router-dom';
import { IGuardRouteProps } from './types';

export function GuardRoute({ canActivate, redirectTo, children }: IGuardRouteProps) {
    if (!canActivate) {
        return <Navigate to={redirectTo} />;
    }

    return children;
}
