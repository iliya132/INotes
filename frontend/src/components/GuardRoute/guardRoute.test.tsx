/* eslint-disable no-undef */
import React from 'react';
import { createRoot, Root } from 'react-dom/client';
import { GuardRoute } from './guardRoute';
import { act } from 'react-dom/test-utils';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

//#region common setup
let root: Root | null = null;
let container: Element | null = null;
beforeEach(() => {
    act(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        root = createRoot(container);
    });
});

afterEach(() => {
    act(() => {
        root!.unmount();
        container!.remove();
        container = null;
    });
});
//#endregion

describe('Test GuardRoute component', () => {
    it('can be rendered', () => {
        act(() => {
            root!.render(
                <MemoryRouter>
                    <Routes>
                        <Route
                            path="/"
                            element={
                                <GuardRoute canActivate={true} redirectTo="/false">
                                    <span>test</span>
                                </GuardRoute>
                            }
                        />
                        <Route path="/false" element={<span>false</span>} />
                    </Routes>
                </MemoryRouter>,
            );
        });

        let permittedComponent = container!.querySelector('span');

        expect(permittedComponent!.textContent).toEqual('test');
        expect(permittedComponent).toMatchSnapshot('guardRoute-permitted-snapshot');
    });

    it('should redirect on false', () => {
        act(() => {
            root!.render(
                <MemoryRouter>
                    <Routes>
                        <Route
                            path="/"
                            element={
                                <GuardRoute canActivate={false} redirectTo="/false">
                                    <span>test</span>
                                </GuardRoute>
                            }
                        />
                        <Route path="/false" element={<span>false</span>} />
                    </Routes>
                </MemoryRouter>,
            );
        });

        let permittedComponent = container!.querySelector('span');

        expect(permittedComponent!.textContent).toEqual('false');
        expect(permittedComponent).toMatchSnapshot('guardRoute-restricted-snapshot');
    });
});
