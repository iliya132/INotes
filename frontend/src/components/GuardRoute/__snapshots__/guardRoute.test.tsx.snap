// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GuardRoute component can be rendered: guardRoute-permitted-snapshot 1`] = `
<span>
  test
</span>
`;

exports[`Test GuardRoute component should redirect on false: guardRoute-restricted-snapshot 1`] = `
<span>
  false
</span>
`;
